### ISSUE TITLE
NOTSET attribute does not supress printing log messages.

### ISSUE THREAD

**Issue by jcoreysc** [adafruit/Adafruit_CircuitPython_Logging#35]
According to the Documentation using the NOTSET attribute should prevent printing of log records. It does not.

I am using Circuit Python 7.3.1 and adafruit_logging version 4.1.1 on a Raspberry PI Pico.

Here is my code:

import adafruit_logging as logging

logger = logging.getLogger("test")
print_logger = logging.StreamHandler()
logger.addHandler(print_logger)

logger.setLevel(logging.NOTSET)
logger.info("Info message\n")
logger.error("Error message\n")

**Comment by FoamyGuy** [comment]
I think one issue is with our documentation. From the CPython docs here: https://docs.python.org/3/library/logging.html#logging.Handler.setLevel it notes:
```
When a handler is created, the level is set to NOTSET (which causes all messages to be processed).
```

I ran your test code adapted for CPython on a PC and it does output the error message:
```
import logging
logger = logging.getLogger("test")
print_logger = logging.StreamHandler()
logger.addHandler(print_logger)

logger.setLevel(logging.NOTSET)
logger.info("Info message")
logger.error("Error message")
```
output:
```
❯ python cpython_test.py 
Error message
```

However the default behavior of effective levels seems slightly different on CPython currently. I'm not quite understanding the hierarchy that it uses to decide which level is the effective level. But it seems that it is coming out to level `30` or `WARNING` in CPython:
```
import logging
logger = logging.getLogger("test")
print_logger = logging.StreamHandler()
logger.addHandler(print_logger)

logger.setLevel(logging.NOTSET)
print(f"effective lvl: {logger.getEffectiveLevel()}")
```
output:
```
effective lvl: 30
```

The same code in CircuitPython shows that the effective level is `0` or `NOTSET`:
```
effective lvl: 0
```

With our effective level defaulting to `0` in CircuitPython it is doing as the CPython docs specify and processing all messages. 


One fix we definitely want to make is to update our documentation to match the CPython doc that specifies `NOTSET` will process all messages (rather than suppress all messages as ours currently states). I'll submit a PR with this documentation change 


Another thing we could consider is changing our default behavior to match the effective level coming out as `WARNING` even when the level was set to `NOTSET` with `logger.setLevel(logging.NOTSET)`. But I'm not sure that we would end up having the whole hierarchy that it seems to be using to figure out which level is effective. Maybe we could simply default to `WARNING` instead of `NOTSET` so that the actual behavior would match CPython even though we don't have the whole system that they are using to determine effective level. I'd like to get feedback from others on this idea though. I'm not sure how the CPython one is making the choice that it is for effective level.

**Comment by jcoreysc** [comment]
From a novice end user perspective, your documentation changes clarify and accurately describe the use of the NOTSET attribute. Since opening this issue I found that the use of the NullHandler provides a more efficient way to disable logging. That is how I will proceed in my code.

I am very much a novice at CircuitPython. I will leave the discussion of the internal behavior to those more qualified than me.

**Comment by FoamyGuy** [comment]
Closed by #36
